### ISSUE TITLE
Focus.EV

### ISSUE THREAD

**Issue by Yzar91** [Farigh/pokeclicker-automation#122]
Optimise road for catch Pokémon with an EV<50 whereas it passed status infected to cure.
When he cured, change road for leveling EV of an another Pokémon…

**Comment by Farigh** [comment]
Requires filtering hatchery priority by EVs, which will be implemented as part of #48

**Comment by Dessastuff** [comment]
Farigh I'd like to see it run dungeons until they're resistant too? is that going to work with hatchery as well?

**Comment by Farigh** [comment]
Implementing dungeon would not be as easy but should not be a problem, for sure.

I don't understand the link with hatchery though...

**Comment by heyboots** [comment]
Yeah you shouldn't need the hatchery at all for this right? The pokedex has filtering for Uninfected/Contagious/Resistant so that info must be accessible. The hatchery is only going to give results for level 100 pokemon that are ready to become egg. I guess prioritization based on Breeding efficiency may be useful, but it also may be easier to just have the script look for routes that still have Missing Resistant marked and just chug through those in order, even if it's not the optimal solution.

**Comment by Farigh** [comment]
The next pokeclicker version adds a pretty interesting feature for this issue : [#3805](https://github.com/pokeclicker/pokeclicker/pull/3805)

**Comment by Farigh** [comment]
Suggested buy [this comment](https://github.com/Farigh/pokeclicker-automation/pull/230#issuecomment-1413795363)

> add an option to skip alternate forms for Focus.PokérusCure

**Comment by GasparMassiot** [comment]
It seems to me that the current PokerusCure Focus doesn't take into account UltraBeasts.
My understanding is that once these pokemons are unlocked, the automation will stay on an UltraBeast Route forever.
It would be nice to avoid this and maybe add an option to capture ultraBeast with beastBalls if available ?

The UltraBeast status can be checked using e.g. `GameConstants.UltraBeastType[pokemons[i]] != undefined`.

Also the current PokerusCure Focus doesn't take roaming pokemons into consideration. Using `RoamingPokemonList` we could add an option to focus on these particular pokemon.

**Comment by Farigh** [comment]
@GasparMassiot do you have a save reproducing the problem with ultrabeasts ?

The roamers is a bit to restrictive to be viable IMO, since it would proc really few times. But why not, I might consider it at some point

**Comment by GasparMassiot** [comment]
[UltraBeast_Block.txt](https://github.com/Farigh/pokeclicker-automation/files/10753491/UltraBeast_Block.txt)

Here is a save where the UltraBeast Block happens.

I got around the issue by creating my own `findNextRouteForPokerus` function that basically works the same as your `findBestRouteForFarmingType` function but focuses on pokerus. 
I can share it if you are interested but the spirit is different from the `Focus.PokerusCure` class you created.

I also added a `moveToRoamingRoute` function to farm Pokerus on roaming pokemons.
